src/components/scroll-view/scroll-view.js

```diff
@@ -105,8 +111,17 @@ export default class ScrollView extends Component {
     let lower = Math.max(offset - threshold, 0);
     let upper = Math.min(offset + visibleItems + threshold, items.length - 1);
 
-    // slice the visible items and map them to `children`
-    return items.slice(lower, upper + 1).map((item, i) => {
+    // slice the visible items
+    let visible = items.slice(lower, upper + 1);
+
+    // empty arrays can still have lengths
+    if (!Object.keys(visible).length) {
+      // in case of an empty array, fill it with null values
+      visible = visible.fill(null);
```

Comment:
This looks like a bug in impagination. If the slice isn't there, the result should be an empty array: 
```js
[].slice(5,100) //=> []
```

Reply:
I believe this _is_ a bug in impagination. The data being sliced is there, but slice still sometimes returns an empty array. 

From what I initially found, `state.length` is correct, but it only has getters for `[0-99]` and `[100-199]` even though the read offset was `> 4000`.

`state.getRecord(offset)` returns the correct record, hence this workaround.